scripts/create_ocd_ids.py

```diff
@@ -0,0 +1,144 @@
+
+contests = {"hr": "Haryana", "mh": "Maharashtra"}
+columns = ["id", "name"]
+country = "in"
+election = "Vidhan Sabha"
+write_table = []
+new_file = True
+
+# replace for other countries
+district_replacements = {
+    "district": " ",
+    "Yamuna Nagar": "Yamunanagar",
+    "Gondia": "Gondiya",
+    "Gurugram": "Gurgaon",
+    "Mewat": "Nuh",
+    "Dhamangaon": "Dhamamgaon",
+    "Deolali": "Devlali",
+    "Chikhali": "Chikhli",
+    "Ballabgarh": "Ballabhgarh",
+    "Kalanwali": "Kalawali",
+    "Navapur": "Nawapur",
+    "Sangole": "Sangola",
+    "Yewla": "Yevla",
+    "Jat": "Jath",
+    "Kalvan": "Kalwan",
+    "Kasba Peth": "Kasbapeth"
+}
+
+const_replacements = {
+    "Yamuna Nagar": "Yamunanagar",
+    "Gondia": "Gondiya",
+    "Gurugram": "Gurgaon",
+    "Mewat": "Nuh",
+    "Dhamangaon": "Dhamamgaon",
+    "Deolali": "Devlali",
+    "Chikhali": "Chikhli",
+    "Ballabgarh": "Ballabhgarh",
+    "Charkhi Dadri": "Dadri",
+    "Kalanwali": "Kalawali",
+    "Navapur": "Nawapur",
+    "Sangole": "Sangola",
+    "Yewla": "Yevla",
+    "Jat": "Jath",
+    "Kalvan": "Kalwan",
+    "Kasba Peth": "Kasbapeth"
+}
+
+punc_replacements = {
+    " ": "_",
+    "(": "",
+    ")": "",
+    "-": "_"
+}
+
+def read_csv(csv_file):
+  # returns in-memory list of table rows
+  table = []
+  with open(csv_file, "rb") as f:
+      csv_reader = csv.DictReader(f)
+      for row in csv_reader:
+        table.append(row)
+  return table
+
+def join_table(consts, districts, state, state_abbr):
+  # return joined table
+  new_table = []
+  ht = {}
+  for d_row in districts:
+    if d_row["district"] in district_replacements:
+      district_key = district_replacements[d_row["district"]]
+    else:
+      district_key = d_row["district"]
+    ht[district_key] = d_row["abbreviation"].lower()
+
+  for c_row in consts:
+    # source of truth on district names:
+    # https://affidavit.eci.gov.in/
+    cons_district = c_row["district"]
+    for old, new in district_replacements.items():
+      if old in cons_district:
+        cons_district = cons_district.replace(old, new)
+    cons_district = cons_district.strip()
+
+    if cons_district in ht:
+      c_row["district"] = cons_district
+      c_row["district_abbr"] = ht[cons_district][:2]
+    else:
+      c_row["district_abbr"] = None
+      c_row["district"] = None
+
+    constituency = c_row["constituency"]
+    if c_row["constituency"] in const_replacements:
+      constituency = const_replacements[c_row["constituency"]]
+    for old, new in punc_replacements.items():
+      constituency = constituency.replace(old, new)
+    c_row["constituency"] = constituency.lower()
+    c_row["state"] = state
+    c_row["state_abbr"] = state_abbr
+    c_row["district"] = cons_district
+    new_table.append(c_row)
+  return new_table
+
+def write_to_file(table):
+  # format hardcoded OCD ID
+  global new_file
+  ocd_id = "ocd-division/country:{}/state:{}/district:{}/cd:{}"
+  rest = "state {} district {} {} constituency {}"
```

Comment:
The naming here is a bit awkward. Maybe

${constituency} constituency, ${district} district, ${state}

E.g.,

Khanapur constituency, Sangli district, Maharashtra

Reply:
Will do. I think I'll use semicolons in lieu of the commas here, unless we should add new column names for constituency, district, and state (can we do that? It could potentially serve a future purpose)